stubs/Application/UI/Multiplier.stub

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Nette\Application\UI;
+
+/**
+ * @template T of \Nette\ComponentModel\IComponent
+ */
+final class Multiplier extends Component
+{
+    /**
+     * @param callable(string, self<T>) : T|callable(string) : T $factory
```

Comment:
Can we drop the union if we mark the 2nd parameter as optional with `=`?

Reply:
Would not it require callable signatire to also always have optional second parameter?

Reply:
Try it ;) phpstan.org/try

Reply:
Ok it works. Updated